File: Structural Patterns/Adapter/Adapter01.js
// Interface do Cliente
class ITarget{
    request(){

    }
}

class Client{
    constructor(target){
        this.target = target;
    }

    makeRequest(){
        console.log("Fazendo uma requisição");
        this.target.request();
    }
}

class Adaptee {
    specificRequest(){
        console.log("Requisição especifica do Adaptee");
    }
}

// Adaptador

class Adapter{
    constructor(Adaptee){
        this.Adaptee = Adaptee;
    }
    request(){
        this.Adaptee.specificRequest();
    }
}

// Uitlização do Adaptador
const adaptee = new Adaptee();
const adapter = new Adapter(adaptee);
const client = new Client(adapter);

client.makeRequest();

File: Structural Patterns/Adapter/Adapter01.py
# Interface do Cliente
class ITarget:
    def request(self):
        pass

    


class Client:
    def __init__(self, target):
        self.target = target
    

    def makeRequest(self):
        print("Fazendo uma requisição")
        self.target.request()

class Adaptee:
    def specificRequest(self):
        print("Requisição especifica do Adaptee")
    


# Adaptador

class Adapter:
    def __init__(self, Adaptee):
        self.Adaptee = Adaptee
    
    def request(self):
        self.Adaptee.specificRequest()


# Uitlização do Adaptador
adaptee = Adaptee()
adapter = Adapter(adaptee)
client = Client(adapter)

client.makeRequest()